app/examples/similarity/index.js

```diff
@@ -0,0 +1,59 @@
+import { bigram } from '../util/datasets';
+import showComponent from '../util/showComponent';
+import SimilarityGraph from '../../../components/SimilarityGraph';
+
+const sonorantClass = letter => 'aeiou'.indexOf(letter) !== -1 ? 'vowel' : 'consonant';
+
+const generateTable = (freq) => {
+  const alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');
+  let table = [];
+
+  let min = Infinity;
+  let max = -Infinity;
+
+  alphabet.forEach(x => {
+    let entry = {
+      id: x,
+      color: sonorantClass(x),
+      size: -Infinity
+    };
+
+    alphabet.forEach(y => {
+      const value = Math.max(bigram[`${x}${y}`], bigram[`${y}${x}`]);
+
+      if (value < min) {
+        min = value;
+      }
+
+      if (value > max) {
+        max = value;
+      }
+
+      if (value > entry.size) {
+        entry.size = value;
+      }
+      entry[y] = value;
+    });
+
+    table.push(entry);
+  });
+
+  // Rescale the "size" values to a reasonable range of circle radii.
+  const minSize = 5;
+  const maxSize = 20;
+  const rescale = val => (val - min) / (max - min) * (maxSize - minSize) + minSize;
+
+  table.forEach(entry => entry.size = rescale(entry.size));
+  console.log(table);
+
+  return table;
+};
+
+window.onload = () => {
+  showComponent(SimilarityGraph, 'svg', {
```

Comment:
> If the container type is div, then the component constructor will have to create its own SVG element within the container. Is that ok?

Yep - that is what e.g. Vega does all the time in SVG rendering mode.

> Related in a way - if we take the approach of constructing our own SVG, would there be any component at all that doesn't take a div as its containing element? If not, then we should think about dumping the "container type" specifier that currently goes with VisComponents.

The use case we were considering is if we had GeoJS layer components whose parent was a GeoJS container. I agree that without that, there is no current use case for components specifying a container type.